### PR TITLE
Add ComicMono

### PR DESCRIPTION
**Pull request by paulfrische**
#### Description

I've added Comic Mono Font.

#### Requirements / Checklist

- [x] Read the [Contributing Guidelines](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md)
- [x] Verified the license of any newly added font, glyph, or glyph set

#### What does this Pull Request (PR) do?
This PR adds the Comic Mono font family. I couldn't make all changes required to the readme because the all patcher script gave me wrong numbers(0 for everything).

Have a Great Day!

### CODE REVIEW

**Comment by Finii** [comment]
`Read the` [`Contributing Guidelines`](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md) has been ticked, but only 2 of 5 steps done. Additionally patched fonts included, please don't they are automatically generated/patched.

**Comment by Finii** [comment]
Btw, thanks for the PR. This will take some time though.

**Comment by Finii** [comment]
> because the all patcher script gave me wrong numbers(0 for everything).

Maybe you can detail this a bit, so that we can fix it and/or the docs. (Screenshots etc)

**Comment by PythonPizzaDE** [comment]
> `Read the` [`Contributing Guidelines`](https://github.com/ryanoasis/nerd-fonts/blob/-/contributing.md) has been ticked, but only 3 of 5 steps done. Additionally patched fonts included, please don't they are automatically generated/patched.

Oh I am sorry. Misunderstood the guidelines. I can give more details in about 2h.

**Comment by Finii** [comment]
> Oh I am sorry. Misunderstood the guidelines. I can give more details in about 2h.

All good, that was more like a note-to-self comment ;-)
Lets improve the guidelines together :-)

**Comment by PythonPizzaDE** [comment]
I've found out why it wasn't working. I ran the script from the bin/scripts dir because in the guidelines it says `./gotta-patch-em-all-font-patcher\!.sh "" info` (./) and an empty string isn't working too, you need to pass a `.`. I was unsure when editing the combinations section because all numbers where higher than the ones given by the script without removing anything. Output of the script:
```
# All fonts patched to sub-directories in 			'patched-fonts'
# The total number of font typefaces ever patched 		'53'
# The total number of font families patched was 		'609'
# The total number of 'complete' patched fonts created was 	'2436'
# The total number of 'variation' patched fonts created was 	'1436478'
# The total number of patched fonts created was 		'1438914'
```
Have a Great Day!

PS: I am sorry for the late reply, couldn't write an answer sooner.

**Comment by Finii** [comment]
> I was unsure when editing the combinations section because all numbers where higher than the ones given by the script without removing anything.

The numbers are completely broken and I was too lazy to fix that when I noticed it. Well, because I personally think ... noone is interested in the numbers ;-D and they are hard to generate in the CI process where we run multiple patch jobs in parallel. I would not worry about them, they SHOULD be auto-updated on releases and not be the burden of a simple contribution (imho).

> PS: I am sorry for the late reply, couldn't write an answer sooner.
No worries!

I will try to fix / amend the docs according to your problems.

One thing missing, but that is also undocumented :roll_eyes: is that some font metadata has to be put into `bin/scripts/lib/fonts.json`; but I can take care about that.

**Comment by Finii** [comment]
I guess you got the font from here: https://github.com/dtinth/comic-mono-font

(No need to reply unless I'm wrong)

image

**Comment by PythonPizzaDE** [comment]
Got it from his github.io page: https://dtinth.github.io/comic-mono-font
I think it's the same.

**Comment by PythonPizzaDE** [comment]
So I removed all the stuff from patched-fonts in my git repo. Should I squash the commits into one so its cleaner?

**Comment by Finii** [comment]
> So I removed all the stuff from patched-fonts in my git repo. Should I squash the commits into one so its cleaner?

If we do not squash then all the patched fonts will end up in the repo (I mean as git objects; you can always check out at the commit where you added them, so the objects are there). So it (squashing) would certainly be the correct thing to do. (But then, we have so much junk in there :-( )

But in any case, as I mentioned right in the top, this PR will take a longer time because I am rather busy, sorry.

**Comment by PythonPizzaDE** [comment]
I think I've did the right git magic now. No worries, don't want to put you under stress😅. G'day

**Comment by PythonPizzaDE** [comment]
Hi, anything new about this PR? G'day!

**Comment by PythonPizzaDE** [comment]
@Finii

**Comment by paulfrische** [comment]
@Finii anything new? Should I change something?

**Comment by Finii** [comment]
Hej @paulfrische 

Well, I think something went wrong with your commit:

image

3000 files changed :thinking: You seem to have removed everything out of `patched-fonts/`.

Let me try to fix this...

**Comment by Finii** [comment]
Correct commit, force push:

image

**Comment by Finii** [comment]
> @Finii anything new? Should I change something?

I guess this is ok.

The schedule was to have `v2.3.0` in November and `v3.0.0` (where this would fit in) in December or January.

Unfortunately now is already ... almost January and `v2.3.0` is still not out. Worse, there are some stoppes that need to be solved first. See also
* #962 
* #716 

Especially the 'bugs' section in 962 is still rather full. At least some should be solved before release. These turned out to be more severe/complicated than anticipated (for example https://github.com/ryanoasis/nerd-fonts/issues/658#issuecomment-1362940306, fixing one bug reveals the next one, and so forth, frustrating)

And then, at the moment I need to work on my Kids' Christmas and New Year stuff in the time Nerd Fonts leaves me :grimacing: 

In other words: As I said in the beginning this PR will take time. Please bear with me.

**Comment by paulfrische** [comment]
@Finii thank you for the update! I just wasn't sure what `take time` means. Happy new year! (in two days)

**Comment by Finii** [comment]
Wanted to pull this for release v3.0.0

Is there a specific reason you use `Comic Mono` and not `Comic Shanns`?

`Comic Mono` is based on `Comic Shanns v1.0`, while there is already `Comic Shanns v2.0`.
Also `Comic Shanns` is on https://www.programmingfonts.org/#comic-shanns

I would just exchange the stuff if you do not disagree?

**Comment by Finii** [comment]
Or probably this https://github.com/jesusmgg/comic-shanns-mono as it is actively maintained

**Comment by paulfrische** [comment]
I've never heard of `Commic Shanns` but looks good to me (so I certainly wouldn't disagree :) ).
Have a Great Day!

**Comment by Finii** [comment]
Just finished the release, lets see if I can cramp it in, still....
(Have not pushed the release yet, so it's only local)

**Comment by Finii** [comment]
Rebase on current release workflow, force push.

**Comment by Finii** [comment]
Adopting ComicShanns was more an issue than expected.

Issues found (and fixed):

`ComicShannsMono` (@jesusmgg): Fix monospace, fix line metrics
`CosmicMono` (@dtinth): Fix self-intersecting glyphs


Also wanting to thank @shannpersand for the upstream upstream source `ComicShanns`.
Thanks @paulfrische for bringing this up.

This will go into NerdFonts `v3.0.0` and therefor will also get a Cask package (or how that is called), because I seem to maintain that also. Other package systems might follow.

**Comment by paulfrische** [comment]
Thank you very much for your time and work! Should I close this PR?

**Comment by Finii** [comment]
Is now in `master`, manually merged.
